UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentCallBackService.cs

```diff
@@ -0,0 +1,95 @@
+                {
+                    Links = new Links()
+                    {
+                        ExchangeSetBatchStatusUri = new LinkSetBatchStatusUri { Href = $"{fileShareServiceConfig.Value.BaseUrl}/batch/{scsResponseQueueMessage.BatchId}" },
+                        ExchangeSetFileUri = new LinkSetFileUri { Href = $"{fileShareServiceConfig.Value.BaseUrl}/batch/{scsResponseQueueMessage.BatchId}/files/{fileShareServiceConfig.Value.ExchangeSetFileName}" }
+                    },
+                    ExchangeSetUrlExpiryDateTime = Convert.ToDateTime(scsResponseQueueMessage.ExchangeSetUrlExpiryDate).ToUniversalTime(),
+                    RequestedProductCount = salesCatalogueProductResponse.ProductCounts.RequestedProductCount.Value,
+                    ExchangeSetCellCount = salesCatalogueProductResponse.ProductCounts.ReturnedProductCount.Value,
+                    RequestedProductsAlreadyUpToDateCount = salesCatalogueProductResponse.ProductCounts.RequestedProductsAlreadyUpToDateCount.Value,
+                    RequestedProductsNotInExchangeSet = GetRequestedProductsNotInExchangeSet(salesCatalogueProductResponse)
+                };
+
+                CallBackResponse callBackResponse = new CallBackResponse()
+                {
+                    Specversion = essCallBackConfiguration.Value.SpecVersion,
+                    Type = essCallBackConfiguration.Value.Type,
+                    Source = essCallBackConfiguration.Value.Source,
+                    Id = Guid.NewGuid().ToString(),
+                    Time = DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture),
+                    Subject = essCallBackConfiguration.Value.Subject,
+                    DataContentType = "application/json",
+                    Data = exchangeSetResponse
+                };
+
+                string payloadJson = JsonConvert.SerializeObject(callBackResponse);
+
+                logger.LogInformation(EventIds.ESSPostCallBackRequestStart.ToEventId(), "Post callback request started for BatchId:{BatchId} and _X-Correlation-ID:{CorrelationId}", scsResponseQueueMessage.BatchId, scsResponseQueueMessage.CorrelationId);
+                
+                await callBackClient.CallBackApi(HttpMethod.Post, payloadJson, scsResponseQueueMessage.CallbackUri);
+                
+                logger.LogInformation(EventIds.ESSPostCallBackRequestCompleted.ToEventId(), "Post callback request completed for BatchId:{BatchId} and _X-Correlation-ID:{CorrelationId}", scsResponseQueueMessage.BatchId, scsResponseQueueMessage.CorrelationId);
+                return true;
+            }
+            else 
+            {
+                logger.LogInformation(EventIds.ESSPostCallBackRequestNotSend.ToEventId(), "Post callback request not send for BatchId:{BatchId} and _X-Correlation-ID:{CorrelationId}", scsResponseQueueMessage.BatchId, scsResponseQueueMessage.CorrelationId);
```

Comment:
Kindly change eventid to ExchangeSetCreatedPostCallbackUriNotProvided and message to "Post callback uri was not provided by requestor for successful exchange set creation for batchid...

Reply:
as per the suggestion i have change eventid to ExchangeSetCreatedPostCallbackUriNotProvided and message.